gameplay/src/Material.cpp

```diff
@@ -136,6 +136,11 @@ Material* Material::create(const char* vshPath, const char* fshPath, const char*
 
     material->_currentTechnique = technique;
 
+	if (material->getParameter("u_diffuseTexture"))
```

Comment:
RenderState::getParameter always returns valid MaterialParameter. If there is no parameter with specified name, it's created and returned. Actually, I found only one way of checking whether parameter with some name is used in the material - by checking all uniforms in all Effects that are used in the Material by iterating over all techniques and passes and calling Effect::getUniform.